### PR TITLE
close #105 add system for saving markers

### LINKED ISSUE #105: system for saving markers
Allow players that slot for a commanding role in an upcoming mission to prepare map markers in advance.

* save player's markers in player's profileNamespace
* load via chatcommand --> select from dialog

### PR DESCRIPTION
**Pull request by McDiod**
- dialog based interaction
- select markers, save selected
- select saved set from list, preview, load
- open with chatcommand or from help
- CBA setting for enabling opening of dialog [never, first 5 minutes of mission, always]

ToDo:
- [x] notification for players that have saved sets for the current map when markers can be loaded

### CODE REVIEW

**Comment by McDiod** [comment]
Current status:

https://youtu.be/yOSJz26lBsg

File: addons/saveMarkers/cfgFunctions.hpp [hunk 1]
```diff
@@ -0,0 +1,33 @@
+class cfgFunctions {
```
Comment (AdlerSalbei) [inline addons/saveMarkers/cfgFunctions.hpp]: Why cfgFunctions not PREP?

File: addons/saveMarkers/functions/fn_createMarkerPreview.sqf [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        "_shape",
+        "_size",
+        "_text",
+        "_type",
+        "_channel"
+    ];
+
+    _markerID = format ["_USER_DEFINED #-1/%1/%2",_forEachIndex,_channel];
+
+    _marker = createMarkerLocal [_markerID,_pos];
```
Comment (AdlerSalbei) [inline addons/saveMarkers/functions/fn_createMarkerPreview.sqf]: missing private

File: addons/saveMarkers/functions/fn_createMarkerPreview.sqf [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+        "_dir",
+        "_pos",
+        "_shape",
+        "_size",
+        "_text",
+        "_type",
+        "_channel"
+    ];
+
+    _markerID = format ["_USER_DEFINED #-1/%1/%2",_forEachIndex,_channel];
```
Comment (AdlerSalbei) [inline addons/saveMarkers/functions/fn_createMarkerPreview.sqf]: missing private

File: addons/saveMarkers/functions/fn_loadMarkers.sqf [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+        "_dir",
+        "_pos",
+        "_shape",
+        "_size",
+        "_text",
+        "_type",
+        "_channel"
+    ];
+
+    _markerID = format ["_USER_DEFINED #-1%1%2/%3/%4",_saveName,_mapName,_forEachIndex,_channel];
```
Comment (AdlerSalbei) [inline addons/saveMarkers/functions/fn_loadMarkers.sqf]: missing private

File: addons/saveMarkers/functions/fn_loadMarkers.sqf [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+
+    _marker = createMarker [_markerID,_pos];
```
Comment (AdlerSalbei) [inline addons/saveMarkers/functions/fn_loadMarkers.sqf]: missing private

Comment (AdlerSalbei) [inline addons/saveMarkers/functions/fn_loadMarkers.sqf]:
```suggestion
    private _marker = createMarker [_markerID,_pos];
```

File: addons/saveMarkers/functions/fn_loadMarkers.sqf (hunk 4 above)
Comment (AdlerSalbei) [inline addons/saveMarkers/functions/fn_loadMarkers.sqf]:
```suggestion
    private _markerID = format ["_USER_DEFINED #-1%1%2/%3/%4",_saveName,_mapName,_forEachIndex,_channel];
```

File: addons/saveMarkers/functions/fn_createMarkerPreview.sqf (hunk 3 above)
Comment (AdlerSalbei) [inline addons/saveMarkers/functions/fn_createMarkerPreview.sqf]:
```suggestion
    private _markerID = format ["_USER_DEFINED #-1/%1/%2",_forEachIndex,_channel];
```

File: addons/saveMarkers/functions/fn_createMarkerPreview.sqf (hunk 2 above)
Comment (AdlerSalbei) [inline addons/saveMarkers/functions/fn_createMarkerPreview.sqf]:
```suggestion
    private _marker = createMarkerLocal [_markerID,_pos];
```

File: addons/saveMarkers/functions/fn_updateSavesList.sqf [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+private _ctrlSavesList = _display displayCtrl IDC_SAVESLIST;
+if (isNull _ctrlSavesList) exitWith {};
+
+lnbClear _ctrlSavesList;
+
+private _saveMarkersData = profileNamespace getVariable [QGVAR(saveData),[]];
+{
+    _x params [["_saveAs","ERROR: NO NAME"],["_worldName","ERROR: NO WORLDNAME"]];
+
+    _rowID = _ctrlSavesList lnbAddRow [_saveAs,_worldName];
```
Comment (AdlerSalbei) [inline addons/saveMarkers/functions/fn_updateSavesList.sqf]:
```suggestion
    private _rowID = _ctrlSavesList lnbAddRow [_saveAs,_worldName];
```

File: addons/saveMarkers/cfgFunctions.hpp (hunk 1 above)
Comment (McDiod) [reply addons/saveMarkers/cfgFunctions.hpp]: dunno, we've been using cfgFunctions in most (all?) modules

Comment (AdlerSalbei) [reply addons/saveMarkers/cfgFunctions.hpp]: yeah, maybe time to change?

Comment (nomisum) [reply addons/saveMarkers/cfgFunctions.hpp]: why PREP? whats the benefit? i'm not even using macros as I dont see the advantages.

**Comment by McDiod** [comment]
Yes

**Pull request merged**